### PR TITLE
Add 'setDefaultLocale' API and enable some other features.

### PR DESCRIPTION
**Pull request by pengrongbo**
1. Add 'init' function instead of using flag to mark whether to send
2. Add 'setDefaultLocale' to support source language isn't English.
3. Add 'dist' folder to store the compiled files.
4. Modify the sample application and readme.

### CODE REVIEW

File: sample/src/app/app.module.ts [hunk 1]
```diff
@@ -4,14 +4,21 @@ import { HttpClientModule } from '@angular/common/http';
...
 import { ClarityModule } from '@clr/angular';
-import { VIPModule, VIPService, PatternCategories, LocaleService, VIPLoader, I18nLoader } from '@singleton-i18n/angular-client';
+import {
+    VIPModule,
+    VIPService,
+    PatternCategories,
+    LocaleService,
+    VIPLoader,
+    I18nLoader
+} from '@vip/ngx-vip';
```
Comment (wangyi-e) [inline sample/src/app/app.module.ts]: @singleton-i18n/angular-client

File: src/services/locale.service.ts [hunk 2]
```diff
@@ -5,6 +5,15 @@
 import { Injectable } from '@angular/core';
 import { Subject } from 'rxjs';
 import { isDefined } from '../util';
+import { VIPServiceConstants } from '../constants';
+
+export interface VLocale {
+    readonly languageCode: string;
```
Comment (wangyi-e) [inline src/services/locale.service.ts]: Only the languageCode is required, right?

Comment (pengrongbo) [reply src/services/locale.service.ts]: Not exactly, all the fields are required.

File: sample/src/app/app.module.ts (hunk 1 above)
Comment (pengrongbo) [reply sample/src/app/app.module.ts]: Fixed

File: sample/src/app/app.module.ts [hunk 3]
```diff
@@ -62,7 +69,22 @@ export function initVIPConfig(service: VIPService, localeService: LocaleService)
     bootstrap: [AppComponent]
 })
 export class AppModule {
-    // constructor(localeService: LocaleService) {
-    //     localeService.setCurrentLocale('zh-Hans','CN');
+    // constructor(localeService: LocaleService, service: VIPService) {
```
[the remark this answers is not in the document]
Comment (pengrongbo) [reply sample/src/app/app.module.ts]: One the other way to initialize i18n resource, just for the reference.

File: sample/src/app/app.module.ts [hunk 4]
```diff
@@ -4,14 +4,21 @@ import { HttpClientModule } from '@angular/common/http';
...
+    I18nLoader
+} from '@vip/ngx-vip';
 import { ENGLISH } from './source/app.l10n';
 import { L10nModule } from './modules/l10n/l10n.module';
 import { I18nModule } from './modules/i18n/i18n.module';
 import { VIPHeaderModule } from './modules/vip-header/vip-header.module';
 
 export function initVIPConfig(service: VIPService, localeService: LocaleService) {
-    localeService.setCurrentLocale('zh-Hans', null, true);
+    localeService.init('zh-Hans');
```
[the remark this answers is not in the document]
Comment (pengrongbo) [reply sample/src/app/app.module.ts]: It could, but for now, the language setting is decoupled.

**Pull request merged**